### ISSUE TITLE
Great CLI but stopped working 30 minutes ago. No clue why

### ISSUE THREAD

**Issue by memento** [Blutsh/Swish#1]
Hi!
First thanks for your great cli, mate.
It's been working like a charm up until 30 minutes ago.

Maybe swish could print what went wrong as it called the API (error status / error name) ?
I didn't spammed the swisstransfer APi (via Swish) but I did uploaded the same file approx 10 times during the 20 to 30 minutes of my testing. I don't know if the crash comes from a too many request response like from the swisstransfer API or anything else.

Here is what I did:.

$ swish my_archive.zip

```output
Request failed, retrying... (0)
Request failed, retrying... (1)
Request failed, retrying... (2)
thread 'main' panicked at src/swissfiles/mod.rs:253:68:
called `Result::unwrap()` on an `Err` value: Error("trailing characters", line: 1, column: 23)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

When I set the env variable RUST_BACKTRACE to 1 as suggested, re-execute the same command, I get the following trace:

```output
thread 'main' panicked at src/swissfiles/mod.rs:253:68:
called `Result::unwrap()` on an `Err` value: Error("trailing characters", line: 1, column: 23)
stack backtrace:
   0: rust_begin_unwind
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:645:5
   1: core::panicking::panic_fmt
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/panicking.rs:72:14
   2: core::result::unwrap_failed
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/result.rs:1654:5
   3: swish::swissfiles::Swissfiles::new_localfiles
   4: swish::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

When I set the env variable RUST_BACKTRACE to 'full' as suggested, re-execute the same command, I get the following trace:

```output
thread 'main' panicked at src/swissfiles/mod.rs:253:68:
called `Result::unwrap()` on an `Err` value: Error("trailing characters", line: 1, column: 23)
stack backtrace:
   0:     0x7eff01ee68c2 - std::backtrace_rs::backtrace::libunwind::trace::h103e47e00552e141
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/../../backtrace/src/backtrace/libunwind.rs:105:5
   1:     0x7eff01ee68c2 - std::backtrace_rs::backtrace::trace_unsynchronized::h592221613f1fd095
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7eff01ee68c2 - std::sys_common::backtrace::_print_fmt::hf8a30f8299b1bef3
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/sys_common/backtrace.rs:68:5
   3:     0x7eff01ee68c2 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h5a6e3e7cc140e753
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x7eff01f2937c - core::fmt::rt::Argument::fmt::hd62a89729b8d4079
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/fmt/rt.rs:142:9
   5:     0x7eff01f2937c - core::fmt::write::haa8a32127a7b0d75
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/fmt/mod.rs:1153:17
   6:     0x7eff01ee462f - std::io::Write::write_fmt::hc541adb06d0aeac1
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/io/mod.rs:1843:15
   7:     0x7eff01ee6694 - std::sys_common::backtrace::_print::h59dc0d5369ede572
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/sys_common/backtrace.rs:47:5
   8:     0x7eff01ee6694 - std::sys_common::backtrace::print::hbb7b999054d983a7
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/sys_common/backtrace.rs:34:9
   9:     0x7eff01ee7b4b - std::panicking::default_hook::{{closure}}::hd143a60ad2fc5a07
  10:     0x7eff01ee78a3 - std::panicking::default_hook::h2aff2dd57e84e09f
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:292:9
  11:     0x7eff01ee7fed - std::panicking::rust_panic_with_hook::h6811313e71f57da8
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:779:13
  12:     0x7eff01ee7ec2 - std::panicking::begin_panic_handler::{{closure}}::h2eb4991eec554504
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:657:13
  13:     0x7eff01ee6d96 - std::sys_common::backtrace::__rust_end_short_backtrace::h05983ae26e2c8878
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/sys_common/backtrace.rs:171:18
  14:     0x7eff01ee7bf4 - rust_begin_unwind
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:645:5
  15:     0x7eff01a1fed5 - core::panicking::panic_fmt::h49eed7d6e9368ad2
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/panicking.rs:72:14
  16:     0x7eff01a203c3 - core::result::unwrap_failed::hb815ab7de92baab5
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/result.rs:1654:5
  17:     0x7eff01a244cc - swish::swissfiles::Swissfiles::new_localfiles::h7383eb8e295232a8
  18:     0x7eff01a29e41 - swish::main::h0249a69769f8cdcd
  19:     0x7eff01a495e6 - std::sys_common::backtrace::__rust_begin_short_backtrace::h849631075150e31e
  20:     0x7eff01a44245 - std::rt::lang_start::{{closure}}::h2a29b06a6ed725e6
  21:     0x7eff01edf815 - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h95600587cb28d946
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/ops/function.rs:284:13
  22:     0x7eff01edf815 - std::panicking::try::do_call::h19463e8b7e5759dc
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:552:40
  23:     0x7eff01edf815 - std::panicking::try::h9c53f8a8b4478cb2
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:516:19
  24:     0x7eff01edf815 - std::panic::catch_unwind::h1bf2d4ddc0dfe252
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panic.rs:146:14
  25:     0x7eff01edf815 - std::rt::lang_start_internal::{{closure}}::h974ac37711bf3c2b
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/rt.rs:148:48
  26:     0x7eff01edf815 - std::panicking::try::do_call::hd0925bfa231be214
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:552:40
  27:     0x7eff01edf815 - std::panicking::try::h4999557a22e5ef52
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:516:19
  28:     0x7eff01edf815 - std::panic::catch_unwind::hbe1275e05de4c7ab
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panic.rs:146:14
  29:     0x7eff01edf815 - std::rt::lang_start_internal::h14632ddfb850e80e
                               at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/rt.rs:148:20
  30:     0x7eff01a2cdd5 - main
```

Any idea?
(I'm not a RUST developer)

Have a good evening.
Take care!

**Comment by Blutsh** [comment]
Hey @memento 

> I don't know if the crash comes from a too many request response like from the swisstransfer API or anything else.

I came across the same issue while heavily testing my tool, (I did upload many times the same file).

Basically the Infomaniak's servers were answering smth like {40X : "Suspicous traffic"}, I did use a VPN to upload again the same file, and it did work.

So my guess is that they flag your IP when they consider you are kinda abusing their service.

So quick workaround for u now is to change your IP x).

I'll implement "verbose mode" so you'll be able to troubleshoot a bit more.

I'll keep you updated. 

Thanks for the feedback.
